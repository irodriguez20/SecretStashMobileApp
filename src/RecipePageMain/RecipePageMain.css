.RecipePageMain {
  background-color: #f8f3f3;
  height: 100%;
  padding: 10px;
  width: 100%;
}

.RecipePageMain__ingredients,
.RecipePageMain__steps {
  color: #181417;
  max-width: 380px;
}
