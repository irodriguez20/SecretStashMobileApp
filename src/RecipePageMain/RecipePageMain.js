import React, { Component } from 'react';
import Recipe from '../Recipe/Recipe'
import Context from '../Context'
import { findRecipe } from '../recipes-helpers'
import './RecipePageMain.css'


class RecipePageMain extends Component {
    static defaultProps = {
        match: {
            params: {}
        }
    }

    static contextType = Context;


    render() {
        const { recipes = [] } = this.context
        const { recipeId } = this.props.match.params

        const recipe = findRecipe(recipes, recipeId) || { content: '' }

        return (
            <section className='RecipePageMain'>
                <Recipe
                    id={recipe.id}
                    name={recipe.name}
                    timetomake={recipe.timetomake}
                    ingredients={recipe.ingredients}
                    description={recipe.description}
                    steps={recipe.steps}
                    history={this.props.history}

                />
                <section>
                    <div className='RecipePageMain__ingredients'>
                        <h4>Ingredients</h4>
                        <ul><li>{recipe.ingredients}</li></ul>
                    </div>
                </section>
                <section>
                    <div className='RecipePageMain__steps'>
                        <h4>Steps</h4>
                        <blockquote>{recipe.steps}</blockquote>
                    </div>
                </section>
            </section>
        );
    }
}

export default RecipePageMain;