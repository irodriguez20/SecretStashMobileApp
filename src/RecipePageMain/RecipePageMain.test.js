import React from "react";
import RecipePageMain from "./RecipePageMain";
import { shallow } from "enzyme";
import shallowToJson from "enzyme-to-json";

describe(`RecipePageMain component`, () => {
    it("renders a .RecipePageMain by default", () => {
        const wrapper = shallow(<RecipePageMain />);
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it.skip("renders a Recipe with recipe prop", () => {
        const props = {
            match: {
                params: {
                    recipeId: 4
                }
            }
        };
        const context = {
            recipes: [
                {
                    id: 4,
                    name: "Salsa",
                    // folderid: 2,
                    timetomake: "15min",
                    description: 'A twist on my mother-in-laws homemade salsa. This salsa is delicous and so quick and easy to whip up. Youll never be satisfied with store-bought again. Youll even start critiquing the restaurants.',
                    ingredients: '2 Jalape√±os, 2 Seranos, 4 Roma Tomatoes, 1/2 Yellow Onion, Salt to taste, 2 Crushed garlic cloves, 1 bunch of Cilantro',
                    steps: '1. Place the tomatoes and peppers in a pot of water. Let it get to a low boil. Once the skin has split on the tomatoes and the peppers color has faded slightly take the pot off the burner. 2. Spoon out the tomatoes and place them in the blender. Spoon out the peppers and cut the stems off. If you dont want all the seeds, slice the pepper in half and scoop out the seeds. Then place the peppers into the blender. 3. Chop the yellow onion and add it to the blender. 4. Blend until the salsa is smooth and everything is combined. 5. Add salt, blend and taste the salsa. If needed add a bit more salt. 6. Chop a handful of cilantro (taking care not to include stems). Add the cilantro to the blender and pulse until just combined. 7. Pour into a serving dish or tupperware and serve!'
                },
            ]
        };
        const recipe = shallow(<RecipePageMain {...props} />, context).find("Recipe");
        expect(toJson(recipe)).toMatchSnapshot();
    });
});
