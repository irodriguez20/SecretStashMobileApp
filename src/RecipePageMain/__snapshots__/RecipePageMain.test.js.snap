// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipePageMain component renders a .RecipePageMain by default 1`] = `
<section
  className="RecipePageMain"
>
  <Recipe />
  <section>
    <div
      className="RecipePageMain__ingredients"
    >
      <h4>
        Ingredients
      </h4>
      <ul>
        <li />
      </ul>
    </div>
  </section>
  <section>
    <div
      className="RecipePageMain__steps"
    >
      <h4>
        Steps
      </h4>
      <blockquote />
    </div>
  </section>
</section>
`;
